### PR TITLE
agents.md: ask to self-review, shorter comments and pr descriptions

### PR DESCRIPTION
**Pull request by lmolkova**
update agents.md to ask AI to produce terse comments and pr description, also to self-review for common issues

### CODE REVIEW

**Review by eternalcuriouslearner — APPROVED** [review]

File: AGENTS.md [hunk 1]
```diff
@@ -146,6 +150,27 @@ run silently uses the previously installed versions.
...
+
+Once the change implementation is complete and tests pass, review the diff in a **new session**. Look for:
+
+- Test coverage gaps - sync/async variants, error paths, streaming, edge cases. Add the missing
+  tests.
+- Instrumentation of layers not owned by the target library (LLM spans reported by agentic framework).
+- Functional issues
+
+Fix the findings, keeping the PR scoped to the original change; for anything outside that scope,
+suggest an issue or a separate PR. Do not open a PR with unresolved major issues or test gaps.
```
Comment (eternalcuriouslearner) [inline AGENTS.md:172]: question(non-blocking): Couple of questions:

1. It will run the precommit, lint etc right. We don't need to add it explicity here.
2. Do we need to add conformance tests etc here?

**Review by lzchen — APPROVED** [review]

File: AGENTS.md (hunk 1 above)
Comment (lmolkova) [reply AGENTS.md:172]: 1. this is for issues identified during self-review - not all of them are caught by CI / precommit.
2. There is a separate section on this - https://github.com/lmolkova/opentelemetry-python-genai/blob/3ddfa3dfaf95c22cc8757dd95d64995876dd292c/AGENTS.md#conformance-tests

File: AGENTS.md [hunk 2]
```diff
@@ -146,6 +150,27 @@ run silently uses the previously installed versions.
...
 - Whenever applicable, all code changes should have tests that actually validate the changes.
 
+## Code comments
+
+Write for someone reading the code as it is now, with no knowledge of the change that produced it.
+
+- Comment only gotchas that can't be inferred from the code (protocol quirks, ordering
+  requirements, upstream bugs worked around). 1-2 sentences max.
+- Never restate what the code does, describe what it used to do, or justify a change - the last one
+  belongs in the PR description or a PR comment.
```
Comment (trask) [inline AGENTS.md:160]: > describe what it used to do, or justify a change

💯

ugh, I keep having this problem in my PRs

Comment (lmolkova) [reply AGENTS.md:160]: I'm close to putting "NEVER add comments unless I explicitly ask for it" in my personal global agents.md 😡
